x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/configure_elasticsearch_engine.tsx

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiHighlight,
+  EuiPanel,
+  EuiSelectable,
+  EuiSpacer,
+  EuiStepsHorizontal,
+  EuiText,
+  EuiTextAlign,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  ALLOWED_CHARS_NOTE,
+  ENGINE_CREATION_FORM_ENGINE_NAME_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER,
+  ENGINE_CREATION_FORM_TITLE,
+  SANITIZED_NAME_NOTE,
+} from './constants';
+
+import { EngineCreationLogic, EngineCreationSteps } from './engine_creation_logic';
+import { IndexStatusDetails, SearchIndexSelectableOption } from './search_index_selectable';
+
+import './engine_creation.scss';
+
+const renderIndexOption = (option: SearchIndexSelectableOption, searchValue: string) => {
+  return (
+    <>
+      <EuiHighlight search={searchValue}>{option.label ?? ''}</EuiHighlight>
+      <EuiSpacer size="xs" />
+      <IndexStatusDetails option={option} />
+    </>
+  );
+};
+
+export const ConfigureElasticsearchEngine: React.FC = () => {
+  const {
+    aliasName,
+    indicesFormatted,
+    isAliasAllowed,
+    isAliasRequired,
+    isLoading,
+    isLoadingIndices,
+    isSubmitDisabled,
+    name,
+    rawName,
+  } = useValues(EngineCreationLogic);
+  const {
+    loadIndices,
+    setIsAliasAllowed,
+    setAliasName,
+    setCreationStep,
+    setRawName,
+    setSelectedIndex,
+  } = useActions(EngineCreationLogic);
+
+  const onChange = (options: SearchIndexSelectableOption[]) => {
+    const selected = options.find((option) => option.checked === 'on');
+    setSelectedIndex(selected?.label ?? '');
+    if (selected?.alias ?? false) setAliasName('');
+    setIsAliasAllowed(!selected?.alias ?? true);
```

Comment:
While it is quite hard to follow what this function does, I would suggest adding some comments. Trying to understand a statement like this `!selected?.alias ?? true` is not that easy :D